packages/strapi-plugin-content-manager/admin/src/components/RelationPreviewList/index.js

```diff
@@ -0,0 +1,51 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import { get } from 'lodash';
+import { Flex, Padded, Count } from '@buffetjs/core';
+import { useIntl } from 'react-intl';
+
+import { getTrad } from '../../utils';
+import { useContentTypeLayout } from '../../hooks';
+import { Truncate, Truncated } from '../CustomTable/styledComponents';
+import CountWrapper from './CountWrapper';
+
+const RelationPreviewList = ({ name, value }) => {
+  const { formatMessage } = useIntl();
+  const { contentType } = useContentTypeLayout();
+  const mainField = get(contentType, ['metadatas', name, 'edit', 'mainField'], '');
+
+  return Array.isArray(value) ? (
```

Comment:
I am not sure this is the correct logic. I believe that you should use the type that you can get directly from the fieldSchema instead.
```
const isSingle = ['oneWay', 'oneToOne', 'manyToOne'].includes(type)

if (isSingle) {
  return <Truncate><Truncated>{value ? value[mainField] : '-'}</Truncated></Truncate>
```

Reply:
Thanks